app/uk/gov/hmrc/merchandiseinbaggagefrontend/service/CalculationService.scala

```diff
@@ -18,17 +18,37 @@ package uk.gov.hmrc.merchandiseinbaggagefrontend.service
 
 import javax.inject.{Inject, Singleton}
 import uk.gov.hmrc.http.{HeaderCarrier, HttpClient}
-import uk.gov.hmrc.merchandiseinbaggagefrontend.connectors.MerchandiseInBaggageConnector
-import uk.gov.hmrc.merchandiseinbaggagefrontend.model.core.{DeclarationGoods, TaxCalculation, TaxCalculations}
+import uk.gov.hmrc.merchandiseinbaggagefrontend.connectors.CurrencyConversionConnector
+import uk.gov.hmrc.merchandiseinbaggagefrontend.model.calculation.CalculationResult
+import uk.gov.hmrc.merchandiseinbaggagefrontend.model.core.{AmountInPence, DeclarationGoods, TaxCalculation, TaxCalculations}
 
 import scala.concurrent.{ExecutionContext, Future}
 
 @Singleton
-class CalculationService @Inject()(override val httpClient: HttpClient)(implicit ec: ExecutionContext) extends MerchandiseInBaggageConnector {
+class CalculationService @Inject()(override val httpClient: HttpClient)(implicit ec: ExecutionContext) extends CurrencyConversionConnector {
 
   def taxCalculation(declarationGoods: DeclarationGoods)(implicit hc: HeaderCarrier): Future[TaxCalculations] =
     Future.traverse(declarationGoods.goods) { good =>
-      getCalculationResult(good.toCalculationRequest).map{ result =>
+      val code = good.purchaseDetails.currency.currencyCode
+      getConversionRate(code).map { rates =>
+        val rounding = BigDecimal.RoundingMode.HALF_UP
+
+        val rate: BigDecimal = rates.find(_.currencyCode == code).fold(BigDecimal(0))(_.rate)
```

Comment:
with suitable imports 

```....setScale(2, HALF_UP)``` or ```....setScale(2, RoundingMode.HALF_UP)``` might be clearer?

Reply:
I think it's fine as it is, it was initially as setScale(2, RoundingMode.HALF_UP) but after I'd typed it 3 times, I thought I'd assign it to a val and just use the val rounding

Reply:
ok